woocommerce/woocommerce-android
# Activate: Native store creation flow improvements

This PR started mainly as a way to dynamically load eCommerce plan data from the API but it also adds a couple of more improvements. 

- The plan price, name and period is loaded dynamically from the `v1.5/plans` endpoint
- The flow can now recover from a `site already exists` error if the site belongs to the current user (see videos below)
- To align with iOS, there is a new `web_checkout` step in tracking
- To be able implement the point above, the cart checkout logic was moved under the Plan purchase screen, which actually makes more sense
- Improved error screen UI
- Improved the scaling of the new store preview (more zoomed out, without distracting dialogs)
- Refactoring of common code to improve reusability (e.g. error and progress screens)
- `NewStore` data is cleared after the store creation is completed, so that it's ready for a new store creation
- The last navigation step to the Installation screen clears the back stack, so that when a new site is created and a user taps the back button, they're redirected back to the site picker, not the previous store creation step
- A couple of minor UI tweaks

**To test:**
1. Start the store creation flow
2. Keep tapping on Continue until the webview checkout step
3. Tap the back button
4. Then tap on Continue to start the checkout page again
5. Verify that the checkout page is displayed again
6. Complete the purchase flow
7. Verify the following track events are logged:
```
🔵 Tracked: site_creation_step, Properties: {"step":"store_name","is_debug":true}
🔵 Tracked: site_creation_step, Properties: {"step":"domain_picker","is_debug":true}
🔵 Tracked: site_creation_step, Properties: {"step":"store_summary","is_debug":true}
🔵 Tracked: site_creation_step, Properties: {"step":"plan_purchase","is_debug":true}
🔵 Tracked: site_creation_step, Properties: {"step":"web_checkout","is_debug":true}
🔵 Tracked: site_creation_step, Properties: {"step":"store_installation","is_debug":true}
🔵 Tracked: login_woocommerce_site_created, Properties: {"source":"switching_store","url":"{{site_url}}","flow":"native","blog_id":{{site_id}},"is_wpcom_store":false,"is_debug":true}
```

https://user-images.githubusercontent.com/1522856/205019856-b71307f2-32a7-4fdd-b93d-998c96307d59.mp4

https://user-images.githubusercontent.com/1522856/205019932-655d63a4-d175-4ac5-9962-42c14b0a2178.mp4
